johnsonandjohnson/Bodiless-JS
# doc(vital-images): Add API doc for vital Images & tweaks for cards/flowcontainers

## Changes
Bold denotes code changes, rest is doc updates.

- Added the vitalImages API so it generates correctly.
- Tweaked the Card/Flow Container documentation and API doc 
- **Restructured the vital-image/FlowContainer so it can be shadowed and added test case in vital-test** -Code Change
- **Update Vital-Image documentation** -Code change only in form of imports/exports.
- Add some documentation to gatsby-theme-bodiless that was relevant to vital-image doc.
- Moved ImageConfiguration to bodiless level and updated it.
- Moved Tailwind Guide to one level higher in site build guide.
- **Fixed bug in WithContentLibrary by removing the ...Default.** -Code Change

## Test Instructions

- Light Smoke on testing images and gatsby images on vital-demo. (confirm import/exports were done correctly)
- On vital-demo, light smoke test on adding images components via flowcontainer.  (restructured to be shadowable)
   - Shadow in vital-test package and added plain square & plain landscape variation.
- Light Smoke test on flowcontainer content library
   - Shadow in vital-test package that all flowcontainers use content library
- Light smoke test on /styleguide/product/
   - Shadow in vital-test pacakge that product description only offers the 3 components: image/youtube/rte and constrained to 100% and a content library button.  If you don't see all components in productdescription this verifies the bug in WithContentLibrary was addressed.
- Check documentation
    - http://localhost:8000/___docs/#/Development/API/@bodiless/vital-image/README
    - http://localhost:8000/___docs/#/VitalDesignSystem/Components/VitalImage/ 
    -  http://localhost:8000/___docs/#/VitalDesignSystem/Components/VitalFlowContainer
    - http://localhost:8000/___docs/#/Components/Image/ImageConfigurations
    - http://localhost:8000/___docs/#/Development/Guides/BuildingSites/TailwindGuide

- Tweaks in documentation
    - http://localhost:8000/___docs/#/VitalDesignSystem/Components/VitalCard
    - http://localhost:8000/___docs/#/Development/API/@bodiless/vital-card/README
    - http://localhost:8000/___docs/#/Development/API/@bodiless/vital-flowcontainer/README
    - http://localhost:8000/___docs/#/Development/API/@bodiless/gatsby-theme-bodiless/enums/GatsbyImagePresets
    - http://localhost:8000/___docs/#/Development/API/@bodiless/gatsby-theme-bodiless/README?id=functions-1
